floodviz/utils.py

```diff
@@ -0,0 +1,82 @@
+import requests
+
+
+def parse_rdb(endpoint, params):
+    """
+        Takes a web service endpoint, a dictionary of key/value pairs for the query parameters
+        and returns all parsed data from the webservice.
+
+        ARGS:
+            endpoint - String specifiying the web service url
+            params - dictionary holding url parameters
+
+                possible keys:
+                    'sites' (mandatory) - List of string site_ids
+                    'site_query' (mandatory) - service specific string for url construction that targets the sites.
+                    'extra' - service specific string prepended to the format if needed.
+                    'start_date_query' - service specific string for url construction that targets the start date of data
+                    'start_date' (madatory with start_date_query) - String specifying the start date for data. (YYYY-MM-DD)
+                    'end_date_query' - service specific string for url construction that targets the end date of data
+                    'end_date' (madatory with end_date_query) - String specifying the end date for data. (YYYY-MM-DD)
+                    'site_status' - service specific string for url construction. Cannot be used with 'agency' key
+                    'agency' - service specific string for url construction. Cannot be used with 'site_status' key
+
+        RETURNS:
+            url - url to request for data
+    """
+    all_data = []
+    sites = None
+    url = endpoint
+    data_format = '?format=rdb'
+    if params.get('sites') is not None:
```

Comment:
My intention was to have the caller know what query parameters to use so that this is truly and general purpose utility. The params should be passed directly to the requests.get after adding the format query parameter so that the output is rdb. We shouldn't be doing a check here to see if the query parameters are valid or whether they make sense.